packages/cli/src/commands/add/createChangeset.ts

```diff
@@ -108,106 +164,40 @@ export default async function createChangeset(
       allPackages
     );
 
-    let pkgJsonsByName = new Map(
+    let pkgJsonsByName: PackageNameJsonMap = new Map(
       allPackages.map(({ packageJson }) => [packageJson.name, packageJson])
     );
 
     let pkgsLeftToGetBumpTypeFor = new Set(packagesToRelease);
 
-    let pkgsThatShouldBeMajorBumped = (
-      await cli.askCheckboxPlus(
-        bold(`Which packages should have a ${red("major")} bump?`),
-        [
-          {
-            name: "all packages",
-            choices: packagesToRelease.map(pkgName => {
-              return {
-                name: pkgName,
-                message: formatPkgNameAndVersion(
-                  pkgName,
-                  pkgJsonsByName.get(pkgName)!.version
-                )
-              };
-            })
-          }
-        ],
-        x => {
-          // this removes changed packages and unchanged packages from the list
-          // of packages shown after selection
-          if (Array.isArray(x)) {
-            return x
-              .filter(x => x !== "all packages")
-              .map(x => cyan(x))
-              .join(", ");
-          }
-          return x;
-        }
-      )
-    ).filter(x => x !== "all packages");
-
-    for (const pkgName of pkgsThatShouldBeMajorBumped) {
-      // for packages that are under v1, we want to make sure major releases are intended,
-      // as some repo-wide sweeping changes have mistakenly release first majors
-      // of packages.
-      let pkgJson = pkgJsonsByName.get(pkgName)!;
+    for (const [i, bump] of bumpTypes.entries()) {
+      const packages = [...pkgsLeftToGetBumpTypeFor];
+      const isLast = i === bumpTypes.length - 1;
+      let pkgsForThisBumpType: string[] = [];
 
-      let shouldReleaseFirstMajor = await confirmMajorRelease(pkgJson);
-      if (shouldReleaseFirstMajor) {
-        pkgsLeftToGetBumpTypeFor.delete(pkgName);
+      if (!packages.length) break;
 
-        releases.push({ name: pkgName, type: "major" });
+      if (isLast) {
```

Comment:
Isn't very clear why things are different for the last bump type vs others to me. Could you explain?